### PR TITLE
Deus Ex Update

### PR DESCRIPTION
**Pull request by N3X15**
# Changes
- Added: Lungs are now fully modular and actually process gas for humans. Lungs must be compatible with a species or they will die of suffocation.
- Added: Vox and Plasmamen have their own lungs. These alien lungs are not available at organ printers, requiring stockpiling from donor monkey-humans.
- Added: All species currently exhale CO2. Vox may be tweaked later to exhale N2O.
- Added: Robotics can make Advanced Lungs (programming 2, biomech 4), which are compatible with all species and filter out some gasses (N2O and plasma), up to certain partial pressures.
# Sprites

Lungs:
Imgur

Eyes:
Imgur
# TODO
- [x] Livers
- [x] Eyes

### CODE REVIEW

**Comment by SkowronX** [comment]
Aw fug we github again?

**Comment by Kurfursten** [comment]
>Oxygen is toxic to Vox even in very small amounts
>But they have Oxygen in their system that bonds with Nitrogen when exhaled

That's silly. Otherwise, fine work.

**Comment by Iamgoofball** [comment]
So did gitlab die

**Comment by Iamgoofball** [comment]
Also: mutons are still in?

**Comment by Probe1** [comment]
:+1: yes please

@Iamgoofball yeah mutons were always in as admin spawn.

**Comment by Fox-McCloud** [comment]
https://github.com/tgstation/-tg-station/pull/8190

You may find this interesting, as it fits in line with what you're doing here.

**Comment by Probe1** [comment]
The sec and med huds look really really nice.  But it'd have to go along with an update to sechuds showing who changed someones arrest status because it's already a big enough pain in the ass when random assitant #998 gets a pair and sets everyone to arrest with impunity, knowing we'll never find out who did it as long as he wears a mask.

**Comment by Fox-McCloud** [comment]
You could do what TG did, and require that you have security access to be able to set the records.

This can be overriden by tators by emagging the glasses or EMPing them.

**Comment by Probe1** [comment]
That would be nice @Fox-McCloud

**Comment by Kurfursten** [comment]
Making Vox exhale oxygen is extremely weird because oxygen is completely poisonous to them.

**Comment by Probe1** [comment]
dunno CO2 is entirely poisonous to us.

**Comment by Kurfursten** [comment]
CO2 is asphyxiant, not toxic.

**Comment by Angelite191** [comment]
I was told to leave feedback here but after following the series of comments, I don't know what I should be leaving feedback on.

**Comment by Probe1** [comment]
Notes so far

- Vox with advanced lungs can breathe normal 80/20 atm.  They still require internals and can ONLY wear nitrogen internals in a gasless environment.

- Advanced Eyes (L1) has no printable L2 and does not need to be called (L1)

File: code/modules/organs/internal/eyes/eyes.dm [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	removed_type = /obj/item/organ/eyes/muton
+
+///////////////
+// BIONIC EYES
+///////////////
+
+/datum/organ/internal/eyes/adv_1
+	name = "advanced eyes"
+	welding_proof=1
+	see_in_dark=5
```
Comment (Probe1) [inline code/modules/organs/internal/eyes/eyes.dm]: does this work?  
imperial_complex_vi_2015-10-14_19-49-36

**Comment by Angelite191** [comment]
I may be retarded, but I still don't get it.

**Comment by Probe1** [comment]
CO2 isn't being added but oxygen is being consumed.

imperial_complex_vi_2015-10-14_20-03-01

**Comment by Probe1** [comment]
You can give a human vox lungs but some snowflaky shit that applied to the advanced lungs applies to humans.  It checks for an OXYGEN TANK only and they are fucked.

imperial_complex_vi_2015-10-14_20-22-22

**Comment by Probe1** [comment]
W
H
E
N

E G G M A N

**Comment by Imagebard** [comment]
>These alien lungs are not available at organ printers, requiring stockpiling from donor monkey-humans.

**NEXIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIS!**
How the fuck do plasmamen even get lungs, they don't have previous species unlike Humans(monkeys) and Vox(chicken).
Only excuse for this is the robotic lungs which can be researched easy.

**Comment by unid15** [comment]
RIP

**Comment by Angelite191** [comment]
Heart status: BROKEN

**Comment by Kurfursten** [comment]
ogre

**Comment by ihadtoregisterforthis** [comment]
Farewell eggman of coding

**Comment by ghost** [comment]
Who's a bad enough dude to take this up?

File: code/modules/organs/internal/lungs/lung.dm [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+	// First, we consume air.
+	for(var/datum/lung_gas/G in gasses)
+		G.set_context(src,breath,H)
+		G.handle_inhale()
+
+	// Next, we exhale. At the moment, only /datum/lung_gas/waste uses this.
+	for(var/datum/lung_gas/G in gasses)
+		G.set_context(src,breath,H)
+		G.handle_inhale()
```
Comment (ghost) [inline code/modules/organs/internal/lungs/lung.dm:41]: Wouldn't this be handle_exhale, considering the comment says that "Next, we exhale"?

**Pull request closed without merging**